### PR TITLE
Refactor Cardano.Tracing.Tracers

### PR DESCRIPTION
**Pull request by Jimbo4350**
- First PR towards making `Cardano.Tracing.Tracers` more easily understandable & explicit.
- All constraints are now explicit,  many functions from within `where` clauses are moved to the top level & given type signatures.
-  `TraceConfig'` is simplified and partial function free.

### CODE REVIEW

File: cardano-config/src/Cardano/TracingOrphanInstances/Network.hs [hunk 1]
```diff
@@ -382,22 +386,10 @@ instance (Show peer)
...
-         , Condense (TxId (GenTx blk))
-         , HasHeader blk
-         , HasTxs blk
-         , HasTxId (GenTx blk)
-         )
+instance ( Condense (HeaderHash blk))
       => ToObject (AnyMessage (BlockFetch blk)) where
-  toObject _verb (AnyMessage (MsgBlock blk)) =
-    mkObject
-      [ "kind" .= String "MsgBlock"
```
Comment (Jimbo4350) [inline cardano-config/src/Cardano/TracingOrphanInstances/Network.hs:394]: I know information was removed here however, the `HasHeader blk` constraint also applies for serialised blocks. It does not make sense to have a `HasHeader` constraint for a serialized block. I'm not entirely sure how we can get around this.

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 2]
```diff
@@ -273,417 +278,479 @@ mkTracers traceConf tracer bcCounters = do
...
+      traceI :: Integral a => Text -> a -> IO ()
+      traceI msg i = traceNamedObject tr' (meta, LogValue msg (PureI (fromIntegral i)))
+
+  traceD "density"     (fromRational density)
+  traceI "slotNum"     slots
+  traceI "blockNum"    blocks
+  traceI "slotInEpoch" (slots `rem` epochSlots)
+  traceI "epoch"       (slots `div` epochSlots)
+ where
+   epochSlots :: Word64 = 21600  -- TODO
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:370]: Please rebase after https://github.com/input-output-hk/cardano-node/pull/1108 is merged.

File: cardano-node/src/Cardano/Node/Run.hs [hunk 3]
```diff
@@ -146,14 +148,19 @@ runNode loggingLayer npm@NodeCLI{protocolFiles} = do
...
                      , ("TraceChainDb",traceChainDB)
                      , ("TraceForge",traceForge)
                      , ("TraceMempool",traceMempool)
                      ]
-            trsinactive = filter (\(_,f) -> not $ traceEnabled (ncTraceConfig nc) f) reqtrs
-        unless (List.null trsinactive) $ do
+
+        trsinactive <- case ncTraceConfig nc of
+                         TracingOff -> return []
+                         TracingOn traceConf -> return $ map (\(tName,f) -> (tName, f traceConf)) reqtrs
```
Comment (CodiePP) [inline cardano-node/src/Cardano/Node/Run.hs:161]: previously, the list only included those tracers which are turned off.

File: cardano-node/src/Cardano/Node/Run.hs [hunk 4]
```diff
@@ -102,17 +100,21 @@ runNode
...
-                           MinimalVerbosity -> "minimal"
-                           MaximalVerbosity -> "maximal"
+    case ncTraceConfig nc of
+      TracingOff -> return ()
+      TracingOn traceConf ->
+        case traceVerbosity traceConf of
+          NormalVerbosity -> traceWith tracer $ "tracing verbosity = normal verbosity "
+          MinimalVerbosity -> traceWith tracer $ "tracing verbosity = minimal verbosity "
+          MaximalVerbosity -> traceWith tracer $ "tracing verbosity = maximal verbosity "
+
```
[the remark this answers is not in the document]
Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Run.hs:117]: Agreed that is more concise. Let me circle back in a follow up PR as I have an idea on how to improve on the above.

File: cardano-node/src/Cardano/Node/Run.hs (hunk 3 above)
Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Run.hs]: My most recent change should now function as before. I'd like to also circle back and propose some further changes here.

**Pull request merged**